reward_extraction/behavior_cloning/behavior_cloning.py

```diff
@@ -0,0 +1,191 @@
+from robot.data import RoboDemoDset
+from robot.base_robot import XArmTaskSpaceEnv
+# export PYTHONPATH="${PYTHONPATH}:/home/xarm/Documents/JacobAndDavin/test/rewardlearning-robot"
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+class BCNet(nn.Module):
+    def __init__(self) -> None:
+        super(BCNet, self).__init__()
+        self.fc1 = nn.Linear(3, 30)
+        self.fc2 = nn.Linear(30, 300)
+        self.fc4 = nn.Linear(300, 30)
+        self.fc3 = nn.Linear(30, 3)
+        self.relu = nn.LeakyReLU()
+    
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.relu(x)    
+        x = self.fc2(x)
+        x = self.relu(x)
+        x = self.fc4(x)
+        x = self.relu(x)
+        x = self.fc3(x)
+
+        return x
+    
+
+def data(SCALE) -> Tuple[DataLoader, DataLoader]:
+    expert_data = RoboDemoDset('/home/xarm/Documents/JacobAndDavin/test/rewardlearning-robot/data/demos/test/demos.hdf', read_only_if_exists=True)
+    state = []
+    targets = []
+    for traj in expert_data:
+        joint_poses = traj['eef_pose']
+        # delta_targets = [] # np.zeros((len(joint_poses)-1, 3))
+        for i in range(0, len(joint_poses) -1):
+            if np.sum(joint_poses[i+1] - joint_poses[i]) == 0:
+                continue
+            state.append(joint_poses[i] / SCALE)
+            targets.append(joint_poses[i+1] / SCALE - joint_poses[i] / SCALE)
+            # targets.append(joint_poses[i]/100)
+        # print(joint_poses[:10])
+        # print(delta_targets[:10])
+
+        # targets.append(targets)
+        
+    state = np.array(state)
+    targets = np.array(targets)
+    perm = torch.randperm(len(state))
+    state = torch.Tensor(state)
+    targets = torch.Tensor(targets)
+    state = state[perm]
+    targets = targets[perm]
+    test_idx = int(len(state) * 0.8)
+    train_x = state[:test_idx]
+    train_y = targets[:test_idx]
+    test_x = state[test_idx:]
+    test_y = targets[test_idx:]
+
+    train_loader = DataLoader(BCDataset(train_x, train_y), batch_size=128, shuffle=True)
+    test_loader = DataLoader(BCDataset(test_x, test_y), batch_size=128, shuffle=True)
+    return train_loader, test_loader
+
+
+
+class BCDataset(Dataset):
+    def __init__(self, x, y) -> None:
+        # super().__init__()
+        self.x = x
+        self.y = y
+    def __len__(self):
+        return(len(self.x))
+
+    def __getitem__(self, idx):
+        return self.x[idx], self.y[idx]
+
+def one_train_iter(net, optimizer, train_loader):
+    critirion = nn.MSELoss()
+    net.train()
+    losses = []
+    for data, labels in train_loader:
+        data, labels = data.to(device), labels.to(device)
+        optimizer.zero_grad()
+        preds = net(data)
+        loss = critirion(preds, labels)
+        losses.append(loss.item())
+        loss.backward()
+        optimizer.step()
+    print(f'Average train loss is {np.mean(losses)}')
+    return (np.mean(losses))
+
+def eval(net, test_loader):
+    critirion = nn.MSELoss()
```

Comment:
doesn't really model for the simple MLP you have above, but you should call model.eval() to put the model in eval mode in case it has something like batch normalization or some sort of distribution sampling